service/handlers.go

```diff
@@ -0,0 +1,185 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/gorilla/mux"
+	"github.com/jackc/pgx/v4"
+	"github.com/mailgun/service/models"
+	"net/http"
+	"os"
+)
+
+const (
+	DomainThreshold  = 1000
+	BouncedThreshold = 1
+)
+
+type BaseHandler struct {
+	db *pgx.Conn
+}
+
+func NewBaseHandler(db *pgx.Conn) *BaseHandler {
+	return &BaseHandler{
+		db: db,
+	}
+}
+
+func (h *BaseHandler) DeliveredHandler() http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+
+		errResp, domainName := getDomain(r)
+
+		err := h.updateOrCreate(domainName, 1, 0)
+		if err != nil {
+			fmt.Println("Error occurred", err)
+			errResp = true
+		}
+
+		response := models.Response{
+			Message:      "Successfully updated domain name",
+			ResponseCode: http.StatusOK,
+			Error:        errResp,
+		}
+		js, err := json.Marshal(response)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+		w.Header().Set("Content-Type", "application/json")
+		_, _ = w.Write(js)
+	}
+}
+
+func (h *BaseHandler) BouncedHandler() http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		errResp, domainName := getDomain(r)
+
+		err := h.updateOrCreate(domainName, 0, 1)
+		if err != nil {
+			errResp = true
+			fmt.Println("Error occurred in BounceHandler", err)
+		}
+
+		response := models.Response{
+			Message:      "Successfully updated domain name",
+			ResponseCode: http.StatusOK,
+			Error:        errResp,
+		}
+		js, err := json.Marshal(response)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+		w.Header().Set("Content-Type", "application/json")
+		_, _ = w.Write(js)
+	}
+
+}
+
+func (h *BaseHandler) GetDomainHandler() http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		errResp, domainName := getDomain(r)
+
+		event, err := h.get(domainName)
+		if err != nil {
+			errResp = true
+			fmt.Println("Error occurred in GetDomainHandler", err)
+		}
+		domainType := determineDomain(event)
+
+		fmt.Println(event, domainType)
+		response := models.GetResponse{
+			Response: models.Response{
+				Message:      "Successfully updated domain name",
+				ResponseCode: http.StatusOK,
+				Error:        errResp,
+			},
+			Event:      event,
+			DomainType: domainType,
+		}
+
+		js, err := json.Marshal(response)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+		w.Header().Set("Content-Type", "application/json")
+		_, _ = w.Write(js)
+	}
+}
+
+func getDomain(r *http.Request) (bool, string) {
+	// get URL parameters
+	vars := mux.Vars(r)
+	errResp := false
+	// find the metric name
+	domainName := vars["domainName"]
+
+	fmt.Println("msg", "Domain name received", "name", domainName)
+
+	return errResp, domainName
+}
+
+func determineDomain(event models.Event) string {
```

Comment:
Could the logic here be made more explicit or simplified?

Reply:
I'm not sure I understand the question here. We could make the method more explicit by sending both the bounce and the delivered params like this 
```
func determineDomain(bounced int, delivered int) string {
```

Reply:
I was thinking something like: https://gist.github.com/roaet/a420a06741fc9d1d386033fb6f7d512d

Reply:
I see. Yes, that looks more clear. I have pushed that change. Thank you for the feedback